src/main/java/com/sap/sgs/phosphor/fosstars/data/github/NvdEntryMatcher.java

```diff
@@ -280,15 +310,74 @@ private boolean parseNodes(List<Node> nodes) {
           continue;
         }
 
-        if (cpeUriCheck(cpeMatch)) {
+        if (projectCheck(cpeMatch, referenceMatch, project)) {
           return true;
         }
       }
 
-      if (parseNodes(node.getChildren())) {
+      if (parseNodes(node.getChildren(), referenceMatch, project)) {
         return true;
       }
     }
     return false;
   }
+
+  /**
+   * Checks if the input {@link URI} is same as the project's URL.
```

Comment:
That's not actually true. The method doesn't check if they are the same. Could you please update the description?